File: comscoreanalytics/src/main/java/com/bitmovin/player/integrations/comscoreanalytics/ComScoreMetadata.java
package com.bitmovin.player.integrations.comscoreanalytics;

import java.util.HashMap;
import java.util.Map;

public class ComScoreMetadata {
    private ComScoreMediaType mediaType;
    private String uniqueContentId;
    private String publisherBrandName;
    private String programTitle;
    private String programId;
    private String episodeTitle;
    private String episodeId;
    private String episodeSeasonNumber;
    private String episodeNumber;
    private String contentGenre;
    boolean advertisementLoad = false;
    private String digitalAirdate;
    private String tvAirdate;
    private String stationTitle;
    private String c3;
    private String c4;
    private String c6;
    boolean completeEpisode = false;
    private String feedType;

    public ComScoreMetadata(ComScoreMediaType mediaType, String uniqueContentId, String publisherBrandName, String programTitle, String programId, String episodeTitle, String episodeId, String episodeSeasonNumber, String episodeNumber, String contentGenre, boolean advertisementLoad, String digitalAirdate, String tvAirdate, String stationTitle, String c3, String c4, String c6, boolean completeEpisode, String feedType) {
        this.mediaType = mediaType;
        this.uniqueContentId = uniqueContentId;
        this.publisherBrandName = publisherBrandName;
        this.programTitle = programTitle;
        this.programId = programId;
        this.episodeTitle = episodeTitle;
        this.episodeId = episodeId;
        this.episodeSeasonNumber = episodeSeasonNumber;
        this.episodeNumber = episodeNumber;
        this.contentGenre = contentGenre;
        this.advertisementLoad = advertisementLoad;
        this.digitalAirdate = digitalAirdate;
        this.tvAirdate = tvAirdate;
        this.stationTitle = stationTitle;
        this.c3 = c3;
        this.c4 = c4;
        this.c6 = c6;
        this.completeEpisode = completeEpisode;
        this.feedType = feedType;
    }

    public Map<String, String> toDictionary() {
        Map<String, String> dictionary = new HashMap<>();
        dictionary.put("ns_st_ci", uniqueContentId);
        dictionary.put("ns_st_pu", publisherBrandName);
        dictionary.put("ns_st_pr", programTitle);
        dictionary.put("ns_st_tpr", programId);
        dictionary.put("ns_st_ep", episodeTitle);
        dictionary.put("ns_st_tep", episodeId);
        dictionary.put("ns_st_sn", episodeSeasonNumber);
        dictionary.put("ns_st_en", episodeNumber);
        dictionary.put("ns_st_ge", contentGenre);

        if (advertisementLoad) {
            dictionary.put("ns_st_ia", "1");
        }

        dictionary.put("ns_st_ddt", digitalAirdate);
        dictionary.put("ns_st_tdt", tvAirdate);
        dictionary.put("ns_st_st", stationTitle);
        dictionary.put("c3", c3);
        dictionary.put("c4", c4);
        dictionary.put("c6", c6);
        if (completeEpisode) {
            dictionary.put("ns_st_ce", "1");
        }
        dictionary.put("ns_st_ft", feedType);

        return dictionary;
    }

    ;

    public ComScoreMediaType getMediaType() {
        return mediaType;
    }
}

File: comscore/src/main/java/com/bitmovin/player/integration/comscore/ComScoreMetadata.kt
package com.bitmovin.player.integration.comscore

class ComScoreMetadata(
    val mediaType: ComScoreMediaType,
    private val uniqueContentId: String? = null,
    private val publisherBrandName: String? = null,
    private val programTitle: String? = null,
    private val programId: String? = null,
    private val episodeTitle: String? = null,
    private val episodeId: String? = null,
    private val episodeSeasonNumber: String? = null,
    private val episodeNumber: String? = null,
    private val contentGenre: String? = null,
    private val advertisementLoad: Boolean = false,
    private val digitalAirdate: String? = null,
    private val tvAirdate: String? = null,
    private val stationTitle: String? = null,
    private val c3: String? = null,
    private val c4: String? = null,
    private val c6: String? = null,
    private val completeEpisode: Boolean = false,
    private val feedType: String? = null
) {

    fun toMap(): Map<String, String?> = mutableMapOf(
        "ns_st_ci" to uniqueContentId,
        "ns_st_pu" to publisherBrandName,
        "ns_st_pr" to programTitle,
        "ns_st_tpr" to programId,
        "ns_st_ep" to episodeTitle,
        "ns_st_tep" to episodeId,
        "ns_st_sn" to episodeSeasonNumber,
        "ns_st_en" to episodeNumber,
        "ns_st_ge" to contentGenre,
        "ns_st_ddt" to digitalAirdate,
        "ns_st_tdt" to tvAirdate,
        "ns_st_st" to stationTitle,
        "c3" to c3,
        "c4" to c4,
        "c6" to c6,
        "ns_st_ft" to feedType
    ).apply {
        if (completeEpisode) {
            put("ns_st_ce", "1")
        }
        if (advertisementLoad) {
            put("ns_st_ia", "1")
        }
    }
}
